server/routerlicious/packages/routerlicious-base/src/alfred/routes/api/api.ts

```diff
@@ -187,7 +187,10 @@ async function checkDocumentExistence(request: Request, storage: core.IDocumentS
     if (!tenantId || !documentId) {
         return Promise.reject(new Error("Invalid tenant or document id"));
     }
-    return storage.getDocument(tenantId, documentId);
+    const document = await storage.getDocument(tenantId, documentId);
+    if (!document || document.deletionTime) {
```

Comment:
I think we need a similar check for getDeltas?
because C1 can independently make an https call for GetDeltas with a valid token for a document.

Reply:
discussed the edge case offline.